library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/GooglePlayServicesLocationProvider.java

```diff
@@ -39,10 +38,8 @@
 
     @Override
     public void onResume() {
-        // not getSourceProvider, because we don't want to connect if it is not already attempt
-        if (!settingsDialogIsOn && googlePlayServicesLocationSource != null &&
-                (isWaiting() || getConfiguration().keepTracking())) {
-            onConnected();
```

Comment:
onResume was used to keep the GoogleClient connection on, but since now that is taken care of by `FusedLocationProviderClient` directly, responsibility here is changed to request a location update so that is guaranteed that location provider client will be connected to this listener. 

[Requesting a location update](https://developers.google.com/android/reference/com/google/android/gms/location/FusedLocationProviderClient#requestLocationUpdates(com.google.android.gms.location.LocationRequest,%20com.google.android.gms.location.LocationCallback,%20android.os.Looper)) multiple times will only replace the existing request, so it is safe to use it for keeping connection here. Any other action onResume can cause multiple calls. For example; Calling `LocationManager.getLocation` in Activity.onCreate will trigger `get` call and after onCreate, `onResume` will be triggered by the system. So, that is why, onResume should only re-attach listeners or re-connect clients nothing more.